codegen/src/attribute.rs

```diff
@@ -108,17 +108,12 @@ impl Step {
         let step_matcher = self.attr_arg.regex_literal().value();
         let caller_name =
             format_ident!("__cucumber_{}_{}", self.attr_name, func_name);
-        let awaiting = if func.sig.asyncness.is_some() {
-            quote! { .await }
-        } else {
-            quote! {}
-        };
-        let unwrapping = match func.sig.output {
-            syn::ReturnType::Default => quote! {},
```

Comment:
Function can return `()` by explicitly specifying it. So we need to match both `fn func(..)` and `fn func(..) -> ()`